bb: allow GOPATH with multiple elements

rminnich opened the pull request:
When I first did this I did not reckon with the full
need we'd have for GOPATH.

This also includes cleanup from some of the gross ways I used
config struct for per-command data; it has a new Command struct
which is used instead.

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

hugelgupf commented on bb/config.go:
```diff
@@ -94,6 +97,7 @@ func doConfig() {
 	if config.Cwd, err = os.Getwd(); err != nil {
 		log.Fatalf("Getwd: %v", err)
 	}
+	config.Gopaths = strings.Split(getenv("GOPATH", ""), ":")
```
Shouldn't this be part of guess go path?

rminnich replied:
yes. Will do.

hugelgupf commented on bb/bb.go:
```diff
@@ -217,22 +221,25 @@ var config struct {
+	// GOPATH is several paths, separated by :
+	// We require that the first element be the
+	// basic path that works with u-root.
+	Gopath  string
```
Why gopath and gopaths? Three should only be one

rminnich replied:
I found a few places where it was useful to have a distinguished path that was purely for u-root commands, so I left it this way. I can be convinced to do something better.

hugelgupf commented on bb/bb.go:
```diff
@@ -300,43 +307,43 @@ func oneFile(dir, s string, fset *token.FileSet, f *ast.File) error {
 	fset := token.NewFileSet()
-	config.FullPath = filepath.Join(config.Gopath, config.CmdPath)
-	p, err := parser.ParseDir(fset, config.FullPath, nil, 0)
+	c.FullPath = filepath.Join(c.Gopath, c.CmdPath)
```
This cold also be in any of the other go paths, no?

rminnich replied:
It's kind of a convenience thing. I almost named this Urootgopath and maybe I should, the intent is that Gopath is where uroot commands are found.

rminnich commented:
Thanks, the reason for gopath vs. gopaths is that gopaths can in principle be empty, and gopath should always contain the gopath for uroot. The name I chose sucks IMHO.

Any other comments are most welcome, I'll wait a little bit and then start addressing your points :-)

rminnich commented:
oops.
